src/core/datastore/sync/sync-manager.js

```diff
@@ -74,6 +74,10 @@ export class SyncManager {
 
   // TODO: this method is temporray, pending fix for MLIBZ-2177
   getSyncItemCountByEntityQuery(collection, query) {
+    if (!query) {
+      return this._syncStateManager.getSyncItemCount(collection);
```

Comment:
Why not call `this.getSyncItemCount(collection)`?

Reply:
`getSyncItemCount` doesn't accept a query, since it can't reliably apply an entity query to the sync queue - the only info about the entities themselves it has is their id. So it accepts a list of entity ids, for which to return the sync items (the only type of entity querying available). So if you pass a query, you have to do a read() with it first and use the ids of matching entities. So that's what's going on.

Reply:
Ok got it